src/transformers/models/bit/modeling_bit.py

```diff
@@ -0,0 +1,911 @@
+        x = self.norm3(x)
+
+        x = self.drop_path(x)
+        x = self.activation(x + shortcut)
+        return x
+
+
+class BitDownsampleConv(nn.Module):
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        stride=1,
+        preact=True,
+        conv_layer=None,
+        norm_layer=None,
+    ):
+        super(BitDownsampleConv, self).__init__()
+        self.conv_layer = conv_layer
+        self.conv = conv_layer(in_channels, out_channels, 1, stride=stride)
+        self.norm = nn.Identity() if preact else norm_layer(num_channels=out_channels, apply_activation=False)
+
+    def forward(self, x):
+        return self.norm(self.conv(x))
+
+
+class BitStage(nn.Module):
+    """
+    A ResNet v2 stage composed by stacked layers.
+    """
+
+    def __init__(
+        self,
+        config,
+        in_channels,
+        out_channels,
+        stride,
+        dilation,
+        depth,
+        bottle_ratio=0.25,
+        layer_dropout=None,
+    ):
+        super().__init__()
+
+        first_dilation = 1 if dilation in (1, 2) else 2
+
+        # Get the layer type
+        if config.layer_type == "bottleneck":
+            layer_fn = BitBottleneckLayer
+        else:
+            layer_fn = BitPreActivationBottleneckLayer
+
+        prev_chs = in_channels
+        self.layers = nn.Sequential()
+        for layer_idx in range(depth):
+            # Get the current hyper-parameters
+            stride, drop_path_rate, is_first_layer = self._get_updated_hyperparameters(
+                layer_idx, stride, layer_dropout
+            )
+
+            self.layers.add_module(
+                str(layer_idx),
+                layer_fn(
+                    config,
+                    prev_chs,
+                    out_channels,
+                    stride=stride,
+                    dilation=dilation,
+                    bottle_ratio=bottle_ratio,
+                    first_dilation=first_dilation,
+                    drop_path_rate=drop_path_rate,
+                    is_first_layer=is_first_layer,
+                ),
+            )
+            prev_chs = out_channels
+            first_dilation = dilation
+
+    def _get_updated_hyperparameters(self, layer_idx, stride, layer_dropout):
+        r"""
+        Get the new hyper-parameters with respect to the previous ones and the index of the current layer.
+        """
+        if layer_dropout:
+            drop_path_rate = layer_dropout[layer_idx]
+        else:
+            drop_path_rate = 0.0
+
+        if layer_idx != 0:
+            stride = 1
+
+        is_first_layer = layer_idx == 0
+
+        return stride, drop_path_rate, is_first_layer
+
+    def forward(self, input: Tensor) -> Tensor:
+        hidden_state = input
+        for _, layer in enumerate(self.layers):
+            hidden_state = layer(hidden_state)
+        return hidden_state
+
+
+class BitEncoder(nn.Module):
+    def __init__(self, config: BitConfig, conv_layer: nn.Module):
```

Comment:
```suggestion
    def __init__(self, config: BitConfig):
```

I don't think `conv_layer` is used no?

Reply:
Fixed in [992d8eb](https://github.com/huggingface/transformers/pull/20550/commits/992d8ebc6f24b4f58b4c7d3df2e8a0cf4145d845)